platforms/vitalam/charts/vitalam-node/templates/statefulset.yaml

```diff
@@ -0,0 +1,173 @@
+                ${VAULT_ADDR}/v1/${vault_secret_key})
+                curl_response=$?
+                if test "$http_code" != "200" ; then
+                  echo "Http response code from Vault - $http_code"
+                  if test "$curl_response" != "0"; then
+                     echo "Error: curl command failed with error code - $curl_response"
+                     exit 1
+                  fi
+                fi
+              fi
+            }
+
+            KUBE_SA_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+            echo "Getting secrets from Vault Server: ${VAULT_ADDR}"
+
+            ## Login to Vault to get an app role token ##
+            VAULT_CLIENT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login \
+              -H "Content-Type: application/json" \
+              -d '{"role":"'"${VAULT_APP_ROLE}"'","jwt":"'"${KUBE_SA_TOKEN}"'"}' | \
+              jq -r 'if .errors then . else .auth.client_token end')
+            validateVaultResponse 'vault login token' "${VAULT_CLIENT_TOKEN}"
+            echo "logged in"
+
+            vault_secret_key="${VAULT_SECRET_PREFIX}/{{ .Values.vault.filename }}"
+            
+            echo "Getting node account keys from $vault_secret_key"
+
+            LOOKUP_SECRET_RESPONSE=$(curl -sS \
+              --header "X-Vault-Token:${VAULT_CLIENT_TOKEN}" \
+              ${VAULT_ADDR}/v1/${vault_secret_key} | \
+              jq -r 'if .errors then . else . end')
+            validateVaultResponse "secret (${vault_secret_key})" "${LOOKUP_SECRET_RESPONSE}" "LOOKUPSECRETRESPONSE" 
+            
+            secretSeed=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["secretSeed"]')
+            echo "${secretSeed}" > secrets/secretSeed
+      containers:
+      - name: node-vitalam
+        image: {{ .Values.images.vitalam }}
+        env:
+        - name: NODE_NAME
+          value: {{ .Values.node.name  }}
+        volumeMounts:
+        - mountPath: /secrets
+          name: keystore
+        - mountPath: {{ .Values.node.basePath }}
+          name: {{ .Values.node.name }}-pv
+        - mountPath: /etc/config
+          name: {{ .Values.node.name }}-genesis-volume
+        command: ["/bin/sh", "-c"]
+        args:
+        - |-
+          #!/bin/sh
+
+          mkdir -p secrets
+
+          ## add configmap data to volume
+          cat /etc/config/genesis.json > secrets/genesis.json
+
+          ## store aura and grandpa keys in node keystore ##
+
+            /vitalam-node/vitalam-node key insert --base-path {{ .Values.node.basePath }} \
+            --chain secrets/genesis.json \
+            --scheme Sr25519 \
+            --suri secrets/secretSeed \
+            --key-type aura
+
+            /vitalam-node/vitalam-node key insert --base-path {{ .Values.node.basePath }} \
+            --chain secrets/genesis.json \
+            --scheme Ed25519 \
+            --suri secrets/secretSeed \
+            --key-type gran
```

Comment:
Is there any reason key insertion is being done at runtime rather than in an init container?  Also consider separating out seeds as there is no guarantee that they will be the same.

Also not sure about the logic of adding your genesis to the keystore as surely this will overwrite our genesis?

Reply:
Will have a look at key insertion part.

And, I believe secret seed remains same for aura and grandpa key for a given node at a given instance. 

Also, Genesis is not being added to node's keystore here, it is being mounted to the container volume to be passed as an argument.